languageserver/integration/commands.go

```diff
@@ -311,243 +372,193 @@ RetryLoop:
-		Type:    protocol.Info,
-		Message: fmt.Sprintf("Submitted transaction id=%s", tx.ID().Hex()),
-	})
-
-	return tx.ID(), nil
+	return
 }
 
-// createAccountHelper creates a new account and returns its address.
-func (i *FlowIntegration) createAccountHelper(conn protocol.Conn) (addr flow.Address, err error) {
-	accountKey := &flow.AccountKey{
-		PublicKey: i.config.ServiceAccountKey.PrivateKey.PublicKey(),
-		SigAlgo:   i.config.ServiceAccountKey.SigAlgo,
-		HashAlgo:  i.config.ServiceAccountKey.HashAlgo,
-		Weight:    flow.AccountKeyWeightThreshold,
-	}
-
-	tx := templates.CreateAccount([]*flow.AccountKey{accountKey}, nil, i.serviceAddress)
+func (i *FlowIntegration) isContractDeployed(address string, name string) (bool, error) {
+	account, err := i.sharedServices.Accounts.Get(address)
 
-	txID, err := i.sendTransactionHelper(conn, i.serviceAddress, tx)
 	if err != nil {
-		return addr, err
+		return false, err
 	}
 
-	// TODO: replace this for loop with a synchronous GetTransaction in SDK
-	// that handles waiting for it to be mined
-	var txResult *flow.TransactionResult
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-	defer cancel()
-	for {
-		txResult, err = i.flowClient.GetTransactionResult(ctx, txID)
-		if err != nil {
-			return addr, err
-		}
-		if txResult.Status == flow.TransactionStatusFinalized ||
-			txResult.Status == flow.TransactionStatusSealed {
-
-			break
-		}
-	}
+	return account.Contracts[name] != nil, nil
+}
 
-	for _, event := range txResult.Events {
-		if event.Type == flow.EventAccountCreated {
-			accountCreatedEvent := flow.AccountCreatedEvent(event)
-			addr = accountCreatedEvent.Address()
-			break
+func (i *FlowIntegration) waitForNetwork() error {
+	// Ping the emulator server for 30 seconds until it is available
+	timer := time.NewTimer(30 * time.Second)
+RetryLoop:
+	for {
+		select {
+		case <-timer.C:
+			return errors.New("emulator server timed out")
+		default:
+			err := i.sharedServices.Status.Ping()
+			if err == nil {
+				break RetryLoop
+			}
 		}
 	}
+	return nil
+}
 
-	if addr == flow.EmptyAddress {
-		return addr, fmt.Errorf("failed to get new account address for tx %s", txID.Hex())
-	}
-
-	i.accounts[addr] = i.config.ServiceAccountKey
-
-	conn.LogMessage(&protocol.LogMessageParams{
+// showMessage will send noti
+func showMessage(conn protocol.Conn, message string) {
+	conn.ShowMessage(&protocol.ShowMessageParams{
 		Type:    protocol.Info,
-		Message: fmt.Sprintf("Created account with address: %s", addr.Hex()),
+		Message: message,
 	})
-
-	return addr, nil
 }
 
-func parseFileFromURI(uri string) string {
-	u, err := url.Parse(uri)
-	if err != nil {
-		return uri
-	}
+func showError(conn protocol.Conn, errorMessage string) {
+	conn.ShowMessage(&protocol.ShowMessageParams{
+		Type:    protocol.Error,
+		Message: errorMessage,
+	})
+}
 
-	return filepath.Base(u.Path)
+func throwErrorWithMessage(conn protocol.Conn, prefix string, err error) error {
```

Comment:
Maybe rename "throw" to "show"

Reply:
That was the first name for it :D
Then I thought - we are returning error here 🤔

How about we just drop prefix and name it `errorWithMessage`?

Reply:
Yep, that sounds great!